app/services/true_layer/api_client.rb

```diff
@@ -16,10 +16,14 @@ def account_holders
     end
 
     def accounts
+      return SimpleResult.new(value: SampleData::ACCOUNTS) if mock_true_layer_data?
```

Comment:
I wonder if it might be worth separating the mock behaviour into a separate object. Something like this:
```ruby
module TrueLayer
  class ApiClientMock < ApiClient
    def accounts
      SimpleResult.new(value: SampleData::ACCOUNTS)
    end

    def transactions(account_id, date_from, date_to)
      SimpleResult.new(value: SampleData::TRANSACTIONS)
    end

    def account_balance(account_id)
      SimpleResult.new(value: SampleData::BALANCES) if mock_true_layer_data?
    end
  end
end
```
Then in `TrueLayer::BankDataImportService`:
```ruby
    def api_client_class
      return ApiClientMock if Setting.mock_true_layer_data?
      ApiClient
    end

    def api_client
      @api_client ||= api_client_class.new(token)
    end
```

Reply:
Great idea